src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3622,16 +3622,22 @@ def _build_identities_info_from_system_user_assigned(cmd, mi_system_assigned, mi
     return identity_types, user_assigned_identities
 
 
-def deallocate_vmss(cmd, resource_group_name, vm_scale_set_name, instance_ids=None, no_wait=False):
+def deallocate_vmss(cmd, resource_group_name, vm_scale_set_name, instance_ids=None, no_wait=False, hibernate=None):
     client = _compute_client_factory(cmd.cli_ctx)
-    if instance_ids and len(instance_ids) == 1:
+    # a walkaround because `VirtualMachineScaleSetVMs#begin_deallocate` does not accept `hibernate`
+    # https://learn.microsoft.com/en-us/rest/api/compute/virtual-machine-scale-set-vms/deallocate?tabs=HTTP
```

Comment:
```suggestion
    # This is a walkaround because the REST service of `VirtualMachineScaleSetVMs#begin_deallocate` does not accept `hibernate` at present
```